app/controllers/users_controller.rb

```diff
@@ -1,22 +1,61 @@
 class UsersController < Clearance::UsersController
+  include CaptchaVerifiable
+  include PrivacyPassSupportable
+
+  before_action :present_privacy_pass_challenge, unless: :valid_privacy_pass_redemption?, only: :new
+
   def new
     @user = user_from_params
   end
 
   def create
     @user = user_from_params
-    if @user.save
-      Mailer.email_confirmation(@user).deliver_later
-      flash[:notice] = t(".email_sent")
-      redirect_back_or url_after_create
+    render template: "users/new" and return unless @user.valid?
+    if !valid_privacy_pass_redemption? && HcaptchaVerifier.should_verify_sign_up?(request.remote_ip)
+      setup_captcha_verification
+      render "users/captcha"
+    elsif @user.save
+      handle_user_after_save
+    else
+      render template: "users/new"
```

Comment:
If I understand it well, this is tricky to test. `valid?` is called few lines before, but this `save` can still fail. In theory codecov could be fixed by using `save!` and `rescue_from`, but I'm not sure if that's worth it. :thinking: